go/apps/api/routes/v2_keys_whoami/handler.go

```diff
@@ -0,0 +1,283 @@
+package handler
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"net/http"
+
+	"github.com/oapi-codegen/nullable"
+	"github.com/unkeyed/unkey/go/apps/api/openapi"
+	"github.com/unkeyed/unkey/go/internal/services/auditlogs"
+	"github.com/unkeyed/unkey/go/internal/services/keys"
+	"github.com/unkeyed/unkey/go/pkg/codes"
+	"github.com/unkeyed/unkey/go/pkg/db"
+	"github.com/unkeyed/unkey/go/pkg/fault"
+	"github.com/unkeyed/unkey/go/pkg/hash"
+	"github.com/unkeyed/unkey/go/pkg/otel/logging"
+	"github.com/unkeyed/unkey/go/pkg/ptr"
+	"github.com/unkeyed/unkey/go/pkg/rbac"
+	"github.com/unkeyed/unkey/go/pkg/vault"
+	"github.com/unkeyed/unkey/go/pkg/zen"
+)
+
+type Request = openapi.V2KeysWhoamiRequestBody
+type Response = openapi.V2KeysWhoamiResponseBody
+
+// Handler implements zen.Route interface for the v2 keys.getKey endpoint
+type Handler struct {
+	// Services as public fields
+	Logger    logging.Logger
+	DB        db.Database
+	Keys      keys.KeyService
+	Auditlogs auditlogs.AuditLogService
+	Vault     *vault.Service
+}
+
+// Method returns the HTTP method this route responds to
+func (h *Handler) Method() string {
+	return "POST"
+}
+
+// Path returns the URL path pattern this route matches
+func (h *Handler) Path() string {
+	return "/v2/keys.whoami"
+}
+
+func (h *Handler) Handle(ctx context.Context, s *zen.Session) error {
+
+	// Authentication
+	auth, err := h.Keys.GetRootKey(ctx, s)
+	if err != nil {
+		return err
+	}
+
+	// Request validation
+	req, err := zen.BindBody[Request](s)
+	if err != nil {
+		return err
+	}
+
+	args := db.FindKeyByIdOrHashParams{
+		ID:   sql.NullString{String: "", Valid: false},
+		Hash: sql.NullString{String: hash.Sha256(req.Key), Valid: true},
+	}
+
+	key, err := db.Query.FindKeyByIdOrHash(ctx, h.DB.RO(), args)
+	if err != nil {
+		if db.IsNotFound(err) {
+			return fault.Wrap(
+				err,
+				fault.Code(codes.Data.Key.NotFound.URN()),
+				fault.Internal("key does not exist"),
+				fault.Public("We could not find the requested key."),
+			)
+		}
+
+		return fault.Wrap(err,
+			fault.Code(codes.App.Internal.ServiceUnavailable.URN()),
+			fault.Internal("database error"),
+			fault.Public("Failed to retrieve Key information."),
+		)
+	}
+
+	// Validate key belongs to authorized workspace
+	if key.WorkspaceID != auth.AuthorizedWorkspaceID {
+		return fault.New("key not found",
+			fault.Code(codes.Data.Key.NotFound.URN()),
+			fault.Internal("key belongs to different workspace"),
+			fault.Public("The specified key was not found."),
+		)
+	}
+
+	// Check if API is deleted
+	if key.Api.DeletedAtM.Valid {
```

Comment:
We can remove this here and in the get key handler since the sql statement already filters out deleted keys and apis